src/test/kotlin/com/amazon/opendistroforelasticsearch/indexstatemanagement/model/XContentTests.kt

```diff
@@ -37,12 +51,41 @@ class XContentTests : ESTestCase() {
         assertEquals("Round tripping State doesn't work", state, parsedState)
     }
 
-    fun `test policy parsing`() {
-        val policy = randomPolicy()
+    fun `test transition parsing`() {
+        val transition = randomTransition()
 
-        val policyString = policy.toJsonString()
-        val parsedPolicy = Policy.parseWithType(parserWithType(policyString))
-        assertEquals("Round tripping Policy doesn't work", policy, parsedPolicy)
+        val transitionString = transition.toJsonString()
+        val parsedTransition = Transition.parse(parser(transitionString))
+        assertEquals("Round tripping Transition doesn't work", transition, parsedTransition)
+    }
+
+    fun `test conditions parsing`() {
+        val conditions = nonNullRandomConditions()
+
+        val conditionsString = conditions.toJsonString()
+        val parsedConditions = Conditions.parse(parser(conditionsString))
+        assertEquals("Round tripping Conditions doesn't work", conditions, parsedConditions)
+    }
+
+    fun `test supplying more than one transition condition fails`() {
```

Comment:
Let's move these (from here and below) tests somewhere else, this file is for XContentTests mostly for parsing whereas these are just validating arguments.

Also can write these as:

```
assertFailsWith(IllegalArgumentException::class, "Expected IllegalArgumentException for multiple transition conditions") {
    Conditions(indexAge = "invalidIndexAge")
}
```

Reply:
Moved argument validation tests to another file called ```ConditionsTests```